packages/esm-patient-forms-app/src/form-entry-interop.ts

```diff
@@ -43,3 +45,35 @@ export function launchFormEntry(
     formInfo: { encounterUuid, formUuid, visit: currentVisit },
   });
 }
+
+/**
+ * Given a list of forms and a list of HtmlFormEntryForm objects from configuration, return a List of HtmlFormEntryForm
+ * returned forms either
+ *  a) have a form resource with a name of `formEngine` and a value of `htmlformentry, or
+ *  b) have an entry in the HtmlFormEntryForm array for a given form uuid
+ * The HtmlFormEntryForm configuration provides a means to override the name and rendering mode of a given form
+ * @param allForms
+ * @param htmlFormEntryForms
+ */
+export function mapFormsToHtmlFormEntryForms(allForms: Array<Form>, htmlFormEntryForms: HtmlFormEntryForm[]) {
+  return allForms
+    ?.filter((form) => {
+      return (
+        htmlFormEntryForms?.some((hfeForm) => hfeForm.formUuid === form.uuid) ||
+        form.resources?.some((resource) => {
+          return resource.name === formEngineResourceName && resource.valueReference === htmlformentryFormEngine;
+        })
+      );
+    })
```

Comment:
The filter takes all of the forms returned from the server, and retains those that either have a server-side resource indicating that they are htmlformentry forms, or a client-side configuration indicating they are htmlformentry forms.